src/lib/coda_commands/coda_commands.ml

```diff
@@ -60,25 +60,20 @@ let is_valid_user_command _t (txn : User_command.t) account_opt =
   in
   Option.is_some remainder
 
-let schedule_user_command t (txn : User_command.t) account_opt :
-    unit Or_error.t Deferred.t =
-  (* FIXME #3457: return a status from Transaction_pool.add and use it instead
-  *)
-  if not (is_valid_user_command t txn account_opt) then
-    return
-      (Or_error.error_string "Invalid user command: account balance is too low")
-  else
-    let logger =
-      Logger.extend
-        (Coda_lib.top_level_logger t)
-        [("coda_command", `String "scheduling a user command")]
-    in
-    let%map () = Coda_lib.add_transactions t [txn] in
-    Logger.info logger ~module_:__MODULE__ ~location:__LOC__
-      ~metadata:[("user_command", User_command.to_yojson txn)]
-      "Submitted transaction $user_command to transaction pool" ;
-    txn_count := !txn_count + 1 ;
-    Or_error.return ()
+let schedule_user_command t (txn : User_command.t) : 'a Or_error.t Deferred.t =
+  let logger =
+    Logger.extend
+      (Coda_lib.top_level_logger t)
+      [("coda_command", `String "scheduling a user command")]
+  in
+  let result_ivar = Ivar.create () in
+  Coda_lib.add_transactions t ([txn], result_ivar) ;
```

Comment:
Curious why Coda_lib.add_transactions doesn't return a Deferred result rather than needing to manually pass in and check the Ivar

Reply:
`add_transactions` doesn't directly apply the transaction to the pool. It writes the transaction to a pipe from which it is written to another pipe and then applied to the pool. The second pipe also receives txns from new blocks and gossip. Also, multiple clients can write to the first pipe. So, you'd lose the information on who sent what. Passing an Ivar seemed like an easy solution.

Reply:
Ah ok, I was wondering whether the ivar needed to be exposed outside of the `Coda_lib.add_transactions` function (since in the 2 uses it seems like we just create a new ivar and pass it in, then read it). Instead add_transactions could create the ivar internally and then return the results of reading the ivar (the deferred). Just curious, there's probably some usecase/nuance that I'm missing here.

Reply:
Ah, yeah. I'll make that change